src/proto/single/cmd.rs

```diff
@@ -42,9 +43,9 @@ pub struct Ack {
 
 impl FaktoryCommand for Ack {
     fn issue<W: Write>(&self, w: &mut dyn Write) -> Result<(), Error> {
-        w.write_all(b"ACK ").map_err(serde_json::Error::io)?;
-        serde_json::to_writer(&mut *w, self)?;
-        Ok(w.write_all(b"\r\n").map_err(serde_json::Error::io)?)
+        w.write_all(b"ACK ").map_err(error::wrap_serde_io)?;
+        serde_json::to_writer(&mut *w, self).map_err(Error::Serialization)?;
+        w.write_all(b"\r\n").map_err(error::wrap_serde_io)
```

Comment:
Do you think that these `w.write_all` should actually be `Error::Serialization`, or should they be `Error::IO`? I made them the former for consistency with the previous results, but I think they should really be the latter since it's more correct and we're breaking anyway.

Reply:
I like the idea of making the `write_all` calls `Error::IO` now!